packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,376 @@
+      if (!cdk.Token.isUnresolved(name)) {
+        const cont = this.props.taskDefinition.node.tryFindChild(name);
+        if (!cont) {
+          throw new Error(`Overrides mention container with name '${name}', but no such container in task definition`);
+        }
+      }
+    }
+
+    this.taskPolicies = this.makePolicyStatements();
+  }
+
+  protected renderTask(): any {
+    if (this.networkConfiguration !== undefined) {
+      // Make sure we have a security group if we're using AWSVPC networking
+      this.securityGroup = this.securityGroup ?? new ec2.SecurityGroup(this, 'SecurityGroup', { vpc: this.props.cluster.vpc });
+      this.connections.addSecurityGroup(this.securityGroup);
+    }
+
+    return {
+      Resource: integrationResourceArn('ecs', 'runTask', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        Cluster: this.props.cluster.clusterArn,
+        TaskDefinition: this.props.taskDefinition.taskDefinitionArn,
+        NetworkConfiguration: this.networkConfiguration,
+        Overrides: renderOverrides(this.props.containerOverrides),
+        ...this.props.launchTarget.bind(this).parameters,
+      }),
+    };
+  }
+
+  private configureAwsVpcNetworking() {
+    const subnetSelection = this.props.subnets ?? { subnetType: this.props.assignPublicIp ? ec2.SubnetType.PUBLIC : ec2.SubnetType.PRIVATE };
+
+    this.networkConfiguration = {
+      AwsvpcConfiguration: {
+        AssignPublicIp: this.props.assignPublicIp ? (this.props.assignPublicIp ? 'ENABLED' : 'DISABLED') : undefined,
+        Subnets: this.props.cluster.vpc.selectSubnets(subnetSelection).subnetIds,
+        SecurityGroups: cdk.Lazy.listValue({ produce: () => [this.securityGroup!.securityGroupId] }),
+      },
+    };
+  }
+
+  private validateLaunchTarget() {
+    if (this.props.launchTarget instanceof EcsFargateLaunchTarget) {
+      this.validateFargateLaunchType();
+    }
+
+    if (this.props.launchTarget instanceof EcsEc2LaunchTarget) {
+      this.validateEc2LaunchType();
+    }
+
+    if (!this.props.taskDefinition.defaultContainer) {
+      throw new Error('A TaskDefinition must have at least one essential container');
+    }
+  }
+
+  private validateEc2LaunchType() {
+    if (!this.props.cluster.hasEc2Capacity) {
+      throw new Error('Cluster for this service needs Ec2 capacity. Call addXxxCapacity() on the cluster.');
+    }
+
+    if (this.props.taskDefinition.networkMode === ecs.NetworkMode.AWS_VPC) {
+      this.configureAwsVpcNetworking();
+    } else {
+      // Either None, Bridge or Host networking. Copy SecurityGroup from ASG.
+      this.validateNoNetworkingProps();
+      this.connections.addSecurityGroup(...this.props.cluster.connections.securityGroups);
+    }
+  }
+
+  private validateNoNetworkingProps() {
+    if (this.props.subnets !== undefined || this.props.securityGroup !== undefined) {
+      throw new Error('vpcPlacement and securityGroup can only be used in AwsVpc networking mode');
```

Comment:
So what should the user do. Which property should they have configured?

Reply:
updated the error message. vpcPlacement was also updated to vpcSubnets a long time ago.